cla_public/apps/base/__init__.py

```diff
@@ -1,5 +1,16 @@
 "Base app"
-from flask import Blueprint
-
+import datetime
+from flask import Blueprint, after_this_request, request
 
 base = Blueprint('base', __name__)
+
+@base.before_app_request
+def detect_user_locale():
+    locale = request.args.get('locale')
+    
+    if locale is not None:
+        @after_this_request
+        def remember_locale(response):
+            expires = datetime.datetime.now() + datetime.timedelta(days=30)
+            response.set_cookie('locale', locale, expires=expires)
```

Comment:
What do you think about removing the current path to set the locale, too? It seems with this changeset we have a duplication of code on setting this cookie.

Reply:
Will do in a separate pull request.